src/components/producerItems/components/ProducerItems.js

```diff
@@ -0,0 +1,127 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { bindActionCreators } from "redux";
+import { connect } from 'react-redux';
+import shallowCompare from 'shallow-equal/objects';
+
+import ProductList from './ProductList';
+import ProductEditModal from './ProductEditModal';
+import { requestProductList, requestProductUpdate } from '../actions';
+import { getProductList } from '../selectors';
+
+/**
+ * React component used to display the list of producer products
+ */
+class ProducerItems extends React.Component {
+    state = {
+        isModalOpen: false,
+        isProductUpdating : false,
+        productToEdit: {},
+    };
+
+    /**
+     * Controls closing the ProductEditModal and resets the state
+     *
+     * @function
+     * @return {Void} - void
+     */
+    closeProductModal = () => {
+        this.setState({
+            isModalOpen: false,
+            isProductUpdating: false,
+            productToEdit: {},
+        });
+    }
+
+    /**
+     * Controls opening the ProductEditModal by setting the `isModalOpen`
+     * and productToEdit values
+     *
+     * @function
+     * @return {Void} - void
+     */
+    openProductModal = id => {
+        const { productList } = this.props;
+        const productToEdit = productList.get(id);
```

Comment:
what does this do

Reply:
`productList` is a [javascript map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) that has all the producer items stored by id. I am simply getting a product item back by passing the list an id.